### PR TITLE
fix: avoid 500 status code with empty http header

### LINKED ISSUE #18: feat: no 500 error if nil header
to update the test case if send an HTTP request with the empty HTTP header.

```lua
=== TEST 7: empty set use_default_user_agent with false
--- http_config eval: $::HttpConfig
--- config
location = /a {
    content_by_lua_block {
        local httpc = require("resty.http").new()
        assert(httpc:connect("127.0.0.1", ngx.var.server_port),
            "connect should return positively")

        local res, err = httpc:request{
            path = "/b",
            headers = {},
            use_default_user_agent = false
        }

        ngx.status = res.status
        ngx.print(res:read_body())

        httpc:close()
    }
}
location = /b {
    content_by_lua_block {
        ngx.say(ngx.req.get_headers()["User-Agent"])
    }
}
--- request
GET /a
--- error_code: 500
--- error_log eval
[qr/invalid value \(boolean\)/]
```

### PR DESCRIPTION
**Pull request by Sn0rt**
fix #18

### CODE REVIEW

File: t/19-user-agent-header.t [hunk 1]
```diff
@@ -222,7 +222,7 @@ bar
 
 
 
-=== TEST 6: empty set use_default_user_agent with true
+=== TEST 6: set use_default_user_agent with true
```
Comment (juststillthinking) [inline t/19-user-agent-header.t]: empty http header and set use_default_user_agent with true

File: t/19-user-agent-header.t [hunk 2]
```diff
@@ -257,7 +257,7 @@ lua-resty-http.*
 
 
 
-=== TEST 7: empty set use_default_user_agent with false
+=== TEST 7: empty http header
```
Comment (juststillthinking) [inline t/19-user-agent-header.t]: empty http header and set use_default_user_agent with false

File: lib/resty/http.lua [hunk 3]
```diff
@@ -329,7 +329,7 @@ local function _format_request(self, params)
         -- Pre-allocate slots for minimum headers and carriage return.
         true,
         true,
-        true,
+        "",
```
Comment (juststillthinking) [inline lib/resty/http.lua:332]: Is this error caused by `table.concat(true, true)`, and why do you just change the last `true`, like `table.concat(true, "")`?

Comment (Sn0rt) [reply lib/resty/http.lua:332]: https://github.com/apache/apisix/actions/runs/4970533577/jobs/8898690402

Comment (juststillthinking) [reply lib/resty/http.lua:332]: Why does the boolean value at index 8 not affect?

**Review by juststillthinking — APPROVED** [review]

File: lib/resty/http.lua (hunk 3 above)
Comment (Sn0rt) [reply lib/resty/http.lua:332]: done

**Comment by Sn0rt** [comment]
@soulbird pls help me take a review

**Review by soulbird — APPROVED** [review]

**Comment by Sn0rt** [comment]
@leslie-tsang pls help me take review

**Pull request merged**